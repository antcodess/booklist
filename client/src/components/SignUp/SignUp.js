import React, { useState } from 'react';
import { connect } from 'react-redux';

import FormInput from '../FormInput/FormInput';
import CustomButton from '../CustomButton/CustomButton';

import { signUpStart } from '../../redux/user/user.actions';
import { setAlert } from '../../redux/alert/alert.actions';

import './SignUp.scss';

export const SignUp = ({ signUpStart, setAlert }) => {
  
  const [userCredentials, setUserCredentials] = useState({
    displayName: '',
    email: '',
    password: '',
    confirmPassword: ''
  });
  
  const { displayName, email, password, confirmPassword } = userCredentials;
  
  const onInputSubmit = (event) => {
    event.preventDefault();
  
    if (password !== confirmPassword) {
      setAlert({
        status: 'failure',
        message: 'Passwords must match.'
      });
      return;
    }
    signUpStart({ email, password, displayName });
  };
  
  const onInputChange = (event) => {
    const { name, value } = event.target;
    setUserCredentials({...userCredentials, [name]: value })
  };

  return (
    <section className='sign-up'>
      <h2>SignUp</h2>
      <h3>I do not have an account.</h3>
      <span>Sign up with your email and password</span>
      <form onSubmit={onInputSubmit}>
      <FormInput 
          name='displayName'
          type='text'
          value={displayName}
          onChange={onInputChange} 
          label='Name'
          placeholder='Name' 
          required 
        />
      <FormInput 
          name='email'
          type='email'
          value={email}
          onChange={onInputChange} 
          label='Email'
          placeholder='Email' 
          required 
        />
        <FormInput 
          name='password'
          type='password'
          value={password}
          onChange={onInputChange} 
          label='Password'
          placeholder='Password' 
          required 
        />
        <FormInput 
          name='confirmPassword'
          type='password'
          value={confirmPassword}
          onChange={onInputChange} 
          label='Confirm Password'
          placeholder='Confirm Password' 
          required 
        />
        <CustomButton 
        type='submit'
        isRegularSignIn>Sign Up</CustomButton>
      </form>
    </section>
  );
};

export const mapDispatchToProps = dispatch => ({
  signUpStart: userCredentials => dispatch(signUpStart(userCredentials)),
  setAlert: alert => dispatch(setAlert(alert))
});

export default connect(null, mapDispatchToProps)(SignUp);