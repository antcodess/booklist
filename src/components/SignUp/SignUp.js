import React, { useState } from 'react';
import { connect } from 'react-redux';

import FormInput from '../FormInput/FormInput';
import CustomButton from '../CustomButton/CustomButton';

import { signUpStart } from '../../redux/user/user.actions';

import './SignUp.scss';

const SignUp = ({ signUpStart }) => {
  
  const [userCredentials, setUserCredentials] = useState({
    displayName: '',
    email: '',
    password: '',
    confirmPassword: ''
  });
  
  const { displayName, email, password, confirmPassword } = userCredentials;
  
  const onInputSubmit = (event) => {
    event.preventDefault();
  
    if (password !== confirmPassword) {
      alert("Passwords don't match");
      return;
    }
    // console.log({ email, password, displayName })
    signUpStart({ email, password, displayName });
  };
  
  const onInputChange = (event) => {
    const { name, value } = event.target;
    setUserCredentials({...userCredentials, [name]: value })
  };

  return (
    <section className='sign-up'>
      <h2>SignUp</h2>
      <h3>I do not have an account.</h3>
      <span>Sign up with your email and password</span>
      <form onSubmit={onInputSubmit}>
      <FormInput 
          name='displayName'
          type='text'
          value={displayName}
          onChange={onInputChange} 
          label='Name'
          placeholder='Name' 
          required 
        />
      <FormInput 
          name='email'
          type='email'
          value={email}
          onChange={onInputChange} 
          label='Email'
          placeholder='Email' 
          required 
        />
        <FormInput 
          name='password'
          type='password'
          value={password}
          onChange={onInputChange} 
          label='Password'
          placeholder='Password' 
          required 
        />
        <FormInput 
          name='confirmPassword'
          type='password'
          value={confirmPassword}
          onChange={onInputChange} 
          label='Confirm Password'
          placeholder='Confirm Password' 
          required 
        />
        <CustomButton 
        type='submit'
        isRegularSignIn>Sign Up</CustomButton>
      </form>
    </section>
  );
};

const mapDispatchToProps = dispatch => ({
  signUpStart: (userCredentials) => dispatch(signUpStart(userCredentials))
})

export default connect(null, mapDispatchToProps)(SignUp);