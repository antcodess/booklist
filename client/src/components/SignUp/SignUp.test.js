import React from 'react';
import { shallow } from 'enzyme';
import { SignUp } from './SignUp';
import { mapDispatchToProps } from './SignUp';

describe('SignUp Component', () => {
  let wrapper;
  let mockSignUpStart;
  let mockSetAlert;
  let dummyFunction = jest.fn();

  const mockSignUpStartPayload = {
    type: "SIGN_UP_START",
    payload: {
      displayName: 'Sample Name',
      email: 'email@example.com', 
      password: 'password'
    }
  };

  const mockUserCredentials = {
    displayName: 'Sample Name',
    email: 'email@example.com', 
    password: 'password',
  }

  const mockAlert = {
    status: 'failure',
    message: 'Passwords must match.'
  }

  const mockSetAlertPayload = {
    type: "SET_ALERT",
    payload: {
      status: 'failure',
      message: 'Passwords must match.'
    }
  };
  
  beforeEach(() => {

    mockSignUpStart = jest.fn();
    mockSetAlert = jest.fn();

    const mockProps = {
      signUpStart: mockSignUpStart, 
      setAlert: mockSetAlert,
      userCredentials: {
        displayName: 'Sample Name',
        email: 'email@example.com', 
        password: 'password',
        confirmPassword: 'password'
      }
    };
    
    wrapper = shallow(<SignUp { ...mockProps } />);
  });

  it('should render SignUp Component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should dispatch signUpStart action when clicked', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).signUpStart(mockUserCredentials);
    expect(dispatch.mock.calls[0][0]).toEqual(mockSignUpStartPayload);
  });

  it('should dispatch setAlert action when clicked', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).setAlert(mockAlert);
    expect(dispatch.mock.calls[0][0]).toEqual(mockSetAlertPayload);
  });
});
