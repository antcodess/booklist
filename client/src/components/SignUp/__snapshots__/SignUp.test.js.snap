// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Component should render SignUp Component 1`] = `ShallowWrapper {}`;
